quesma/clickhouse/ingest_validator.go

```diff
@@ -56,11 +56,25 @@ func getTypeName(v interface{}) string {
 		if len(elem) == 0 {
 			return arrayLiteral + "(unknown)"
 		} else {
-			return arrayLiteral + "(" + getTypeName(elem[0]) + ")"
+			innerTypeName := getTypeName(elem[0])
```

Comment:
I think that we can extract this piece of code as separate function as use it in `interface{}` case as well

Reply:
Is it OK if I also use reflection (`reflect.ValueOf`) in the case of arrays? Otherwise I don't see a good way to extract common logic (apart from extracting 2-3 lines to a separate function).

Reply:
a ok. I'm not if that will have performance impact and if so, we can leave as it is